app/layout.tsx

```diff
@@ -107,8 +107,7 @@ export default function RootLayout({
 }>) {
     return (
         <ClerkProvider
-            signInUrl="/sign-in"
-            signUpUrl="/sign-up"
+            signInUrl="/enter"
             afterSignOutUrl="/"
             appearance={clerkAppearance}
```

Comment:
Removing `signUpUrl` was **intentional and the entire point of this PR**.

Clerk released a combined sign-in-or-up flow in January 2025. When you remove `signUpUrl`, Clerk's `<SignIn />` component automatically handles both sign-in AND sign-up:

1. User enters non-existent email → Clerk transitions to sign-up form
2. Email is pre-filled (no re-entry)
3. "Already have an account?" → returns to sign-in

The legacy `/sign-up` redirect is for backward compatibility with old links. The actual flow works correctly at `/enter` without any redirect hop.

See: https://clerk.com/changelog/2025-01-16-sign-in-or-up

Tested extensively - this is working as designed.